contrail_api_cli_extra/clean/fip.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from sys import exit
+import argparse
+
+from contrail_api_cli.command import Option, Arg
+from contrail_api_cli.utils import printo
+from contrail_api_cli.resource import Collection, Resource
+from contrail_api_cli.exceptions import ResourceNotFound, CollectionNotFound
+
+from ..utils import server_type, CheckCommand
+from refs import CleanRefs
+
+class CleanBrokenFips(CheckCommand):
+    """
+    """
+    description = "Clean for broken floating IPs"
+
+    pool = Option(help="floating IP pool ref",
+                  nargs="?")
+    fip_refs = Arg(help="list of refs [fip0, fip1, fip2, fip3, ...]",
+                   nargs="*", default=[])
+    cassandra_servers = Option(help="cassandra server list' (default: %(default)s)",
+                               nargs='+',
+                               type=server_type,
+                               default=['localhost:9160'])
+
+    def _clean_refs(self, src, tgt, **kwargs):
+        try:
+            fip = Resource('floating-ip', uuid=tgt, fetch=True)
+        except ResourceNotFound:
+            CleanRefs(ref_type="children",
+                      target_type="floating_ip",
+                      paths=[src, tgt],
+                      **kwargs)
```

Comment:
Regarding fip refs, there is also project -> fip refs that might be broken also and should be removed
Do you think it could be handled by this command ?

Reply:
It will be great but since we cannot find the project from the fip object, do you think we can list only project that have a floating_ip_back_refs field in the contrail API?